### PR TITLE
fix(shadowdom): allow conditional content tags.

### PR DESCRIPTION
**Pull request by rkirov**
Distribution is triggered on the host element injector after each new
view creation.

### CODE REVIEW

File: modules/angular2/src/core/compiler/view_container.js [hunk 1]
```diff
@@ -38,17 +40,20 @@ export class ViewContainer {
...
     this.appInjector = null;
     this.hostElementInjector = null;
+    this.hostLightDom = null;
     this._eventManager = eventManager;
   }
 
   hydrate(appInjector: Injector, hostElementInjector: eiModule.ElementInjector) {
     this.appInjector = appInjector;
     this.hostElementInjector = hostElementInjector;
+    this.hostLightDom = isPresent(hostElementInjector) ? hostElementInjector.get(SourceLightDom) : null;
```
Comment (vsavkin) [inline modules/angular2/src/core/compiler/view_container.js]: Should we rename SourceLightDom into HostLightDom? If not, maybe it is better to call the variable sourceLightDom?

File: modules/angular2/src/core/compiler/view_container.js [hunk 2]
```diff
@@ -19,6 +20,7 @@ export class ViewContainer {
   elementInjector: eiModule.ElementInjector;
   appInjector: Injector;
   hostElementInjector: eiModule.ElementInjector;
+  hostLightDom;
```
Comment (vsavkin) [inline modules/angular2/src/core/compiler/view_container.js]: Type?

**Comment by vsavkin** [comment]
LGTM with a few minor comments

File: modules/angular2/src/core/compiler/view_container.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ import * as eiModule from 'angular2/src/core/compiler/element_injector';
 import {isPresent, isBlank} from 'angular2/src/facade/lang';
 import {EventManager} from 'angular2/src/core/events/event_manager';
 import {Reflector} from 'angular2/src/reflection/reflection';
+import {SourceLightDom} from './shadow_dom_emulation/light_dom';
```
Comment (mlaval) [inline modules/angular2/src/core/compiler/view_container.js]: This creates a circular dependency, you could replace it with:
`import * as ldModule from './shadow_dom_emulation/light_dom';`

File: modules/angular2/test/core/compiler/shadow_dom/shadow_dom_emulation_integration_spec.js [hunk 4]
```diff
@@ -187,33 +187,32 @@ export function main() {
...
-        //
-        //    cmp.hideLeft()
-        //    lc.tick();
-        //
-        //    expect(view.nodes).toHaveText('(, ABC)');
-        //
-        //    done();
-        //  });
-        //});
+        it('should redistribute when the shadow dom changes', (done) => {
```
Comment (mlaval) [inline modules/angular2/test/core/compiler/shadow_dom/shadow_dom_emulation_integration_spec.js]: See #938 about build failures

File: modules/angular2/src/core/compiler/view_container.js (hunk 3 above)
Comment (rkirov) [inline modules/angular2/src/core/compiler/view_container.js]: I am fixing it, but for future reference how can I detect them without manually checking.

File: modules/angular2/src/core/compiler/view_container.js (hunk 2 above)
Comment (rkirov) [inline modules/angular2/src/core/compiler/view_container.js]: done.

File: modules/angular2/test/core/compiler/shadow_dom/shadow_dom_emulation_integration_spec.js (hunk 4 above)
Comment (rkirov) [inline modules/angular2/test/core/compiler/shadow_dom/shadow_dom_emulation_integration_spec.js]: thanks, merged your PR!

File: modules/angular2/src/core/compiler/view_container.js (hunk 1 above)
Comment (rkirov) [inline modules/angular2/src/core/compiler/view_container.js]: In fact as far the injector is concerned this is not HostLightDom but just a LightDom.

**Pull request closed without merging**